corehq/apps/registration/views.py

```diff
@@ -371,9 +374,20 @@ def resend_confirmation(request):
         return redirect('domain_select')
 
     context = get_domain_context()
+    default_page_name = _('Resend Confirmation Email')
 
     if request.method == 'POST':
+        if (datetime.utcnow() - dom_req.request_time).seconds < RESEND_INVITATION_TIME_INTERVAL:
+            context = {
+                'message_body': _(f'Please wait at least {RESEND_INVITATION_TIME_INTERVAL} '
+                                  f'seconds before requesting again.'),
+                'current_page': {'page_name': default_page_name},
+            }
+            return render(request, 'registration/confirmation_error.html', context)
         try:
+            dom_req.request_time = datetime.utcnow()
```

Comment:
will updating the request time impact any other places? I see it is used in a few other places to filter requests.

Reply:
that is a great point.
Its used at two more places,
1. https://github.com/dimagi/commcare-hq/blob/31f4f5989214ea76d437407da25dbd8bbf9a9453/corehq/apps/registration/views.py#L309 , to limit users for the number of domain requests they make in a day which is 99. So if someone requests the link again for a registration done in the past, it would count the request as being done on the current day as well.
2. https://github.com/dimagi/commcare-hq/blob/c0dd17cc35182ae585ab4260925fadceb299aa7e/corehq/apps/registration/tasks.py#L26 to send reminder emails for registrations in past 24 hours, so if someone requests for a link again, after 24 hours of registering the domain, they would be sent another email for activation in next 24 hours.

Both of these look okay to me for update. But also going to tag @dimagi/product to confirm.

(there is also a field by that name on a different model, so a general code search does flag many instances)

Reply:
Thanks for the explanations, those seem OK to me.